azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventHubClient.java

```diff
@@ -56,9 +57,23 @@ private EventHubClient(final ConnectionStringBuilder connectionString) throws IO
 	public static EventHubClient createFromConnectionStringSync(final String connectionString)
 			throws ServiceBusException, IOException
 	{
+		return createFromConnectionStringSync(connectionString, null);
+	}
+	
+	/**
+	 * Synchronous version of {@link #createFromConnectionString(String)}. 
+	 * @param connectionString The connection string to be used. See {@link ConnectionStringBuilder} to construct a connectionString.
+	 * @param retryPolicy A custom {@link RetryPolicy} to be used when communicating with EventHub.
+	 * @return EventHubClient which can be used to create Senders and Receivers to EventHub
+	 * @throws ServiceBusException If Service Bus service encountered problems during connection creation. 
+	 * @throws IOException  If the underlying Proton-J layer encounter network errors.
+	 */
+	public static EventHubClient createFromConnectionStringSync(final String connectionString, final RetryPolicy retryPolicy)
+			throws ServiceBusException, IOException
+	{
 		try
 		{
-			return createFromConnectionString(connectionString).get();
+			return createFromConnectionString(connectionString, retryPolicy).get();
```

Comment:
>retryPolicy [](start = 55, length = 11)

One other way to handle custom RetryPolicies is to allow users to do something like : RetryPolicy.registerRetryPolicy("custom", typeof(CustomRetryPolicy)). Store this in a static property bag and use this when we are parsing connStr.